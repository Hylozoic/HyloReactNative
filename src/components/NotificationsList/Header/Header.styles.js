export default {
  closeButton: {
    marginLeft: 10,
    fontSize: 16
  },
  title: {
    alignSelf: 'center',
    fontFamily: 'Circular-Bold',
    fontSize: 18
  },
  newButton: {
    marginRight: 10,
    fontSize: 16
  }
}
