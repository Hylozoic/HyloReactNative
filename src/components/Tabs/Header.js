import React from 'react'
import { Text } from 'react-native'

import MenuButton from './MenuButton'
import styles from './styles'

export default function Header (navigation) {
  return {
    headerTitle: <Text style={styles.headerTitle}>Home</Text>,
    headerLeft: <MenuButton navigation={navigation} />
  }
}
