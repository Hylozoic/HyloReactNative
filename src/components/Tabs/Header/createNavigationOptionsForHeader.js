import React from 'react'
import { View } from 'react-native'
import MenuButton from './MenuButton'
import SearchIcon from './SearchIcon'
import MessagesIcon from './MessagesIcon'
import NotificationsIcon from './NotificationsIcon'
import styles from './Header.styles'

export default function createNavigationOptionsForHeader (navigation, title) {
  const openDrawer = () => navigation.openDrawer()
  const showSearch = () => navigation.navigate('SearchPage')
  const showMessages = () => navigation.navigate('ThreadList')
  const showNotifications = () => navigation.navigate('NotificationsList')

  return {
    headerBackTitle: null,
    headerTitle: title,
    headerTitleStyle: styles.headerTitle,
    headerLeft: () => 
      <MenuButton openDrawer={openDrawer} />,
    headerRight: () => 
      <View style={styles.controls}>
        <SearchIcon
          showSearch={showSearch} />
        <MessagesIcon
          showMessages={showMessages} />
        <NotificationsIcon
          showNotifications={showNotifications} />
      </View>
  }
}
