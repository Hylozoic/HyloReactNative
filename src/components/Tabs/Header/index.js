import React from 'react'
import { Text, View } from 'react-native'

import MenuButton from './MenuButton'
import MessagesIcon from './MessagesIcon'
import NotificationsIcon from './NotificationsIcon'
import styles from './Header.styles'

export default function Header (navigation, title) {
  const openDrawer = () => navigation.navigate('DrawerOpen')
  const showMessages = () => navigation.navigate('ThreadList')
  const showNotifications = () => navigation.navigate('NotificationsList', { title: 'Notifications' })

  return {
    headerTitle: <Text style={styles.headerTitle}>{title}</Text>,
    headerLeft: <MenuButton openDrawer={openDrawer} />,
    headerRight: <View style={styles.controls}>
      <MessagesIcon showMessages={showMessages} />
      <NotificationsIcon showNotifications={showNotifications} />
    </View>
  }
}
