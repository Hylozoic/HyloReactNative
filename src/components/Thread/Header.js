import React from 'react'
import { StyleSheet, Text } from 'react-native'

export default function Header (navigation) {
  return {
    headerTitle: navigation.state.params.title,
    headerTitleStyle: styles.title
  }
}

const styles = StyleSheet.create({
  title: {
    fontFamily: 'Circular-Bold',
    fontSize: 18,
    marginLeft: 15,

    // Required to avoid font-weight bug where RN goes looking for
    // CustomFontName_bold.ttf, which doesn't exist:
    // https://github.com/react-community/react-navigation/issues/542#issuecomment-283663786
    fontWeight: '200'
  }
})
