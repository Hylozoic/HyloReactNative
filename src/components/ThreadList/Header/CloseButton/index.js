import React from 'react'
import { TouchableOpacity, Text } from 'react-native'

import styles from '../Header.styles.js'

export default function CloseButton (props) {
  const { closeMessages } = props
  return <TouchableOpacity onPress={closeMessages}>
    <Text style={styles.closeButton}>Close</Text>
  </TouchableOpacity>
}
