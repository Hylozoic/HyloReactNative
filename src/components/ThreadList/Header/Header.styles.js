export default {
  closeButton: {
    marginLeft: 10,
    fontSize: 16
  },
  title: {
    alignSelf: 'center',
    fontWeight: '900',
    fontSize: 18
  },
  newButton: {
    marginRight: 10,
    fontSize: 16
  }
}
