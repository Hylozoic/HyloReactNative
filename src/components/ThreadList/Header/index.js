import React from 'react'
import { Text, TouchableOpacity } from 'react-native'

import CloseButton from './CloseButton'
import styles from './Header.styles.js'

export default function Header (navigation) {
  const { goBack } = navigation

  const closeMessages = () => {
    goBack()
  }

  const goToNewMessage = () => {
    navigation.navigate('NewMessage')
  }

  return {
    headerTitle: <Text style={styles.title}>Messages</Text>,
    headerLeft: <CloseButton closeMessages={closeMessages} />,
    headerRight: <TouchableOpacity
      onPress={goToNewMessage}
      hitSlop={{top: 10, bottom: 10, left: 10, right: 10}}>
      <Text style={styles.newButton}>New</Text>
    </TouchableOpacity>
  }
}
