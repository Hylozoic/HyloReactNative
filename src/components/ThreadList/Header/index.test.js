import {TouchableOpacity} from 'react-native'
import TestRenderer from 'react-test-renderer'
import Header from './index'

describe('headerRight', () => {
  it('goes to new message page', () => {
    const navigation = {
      navigate: jest.fn()
    }
    const renderer = TestRenderer.create(Header(navigation).headerRight)
    expect(renderer.toJSON()).toMatchSnapshot()
    const { root } = renderer
    root.findByType(TouchableOpacity).props.onPress()
    expect(navigation.navigate).toBeCalledWith('NewMessage')
  })
})
