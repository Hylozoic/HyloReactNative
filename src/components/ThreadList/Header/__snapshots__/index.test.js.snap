// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`headerRight goes to new message page 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={
    Object {
      "bottom": 10,
      "left": 10,
      "right": 10,
      "top": 10,
    }
  }
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    New
  </Text>
</View>
`;
